### ISSUE TITLE
srfi-27 with large range parameter not returning all numbers

### ISSUE THREAD

**Issue by arvyy** [ashinn/chibi-scheme#675]
Looks like regression happened of https://github.com/ashinn/chibi-scheme/issues/668 , although this time with bigger numbers

```scheme
(import 
  (scheme base)
  (srfi 27)
  (srfi 158)
  (chibi test))

(test-begin "High number")

(define range (expt 2 64))
(define mid (expt 2 63))

(define int-generator
  (let ((get-int (random-source-make-integers default-random-source)))
   (lambda ()
     (get-int range))))

(test-assert
  (generator-any
    (lambda (int)
      (> int mid))
    (gtake int-generator 10000)))

(test-end "High number")
```

Test fails on newest version from master built with default settings. Trying to build from older commits gives me segfault for some reason, not sure why, so unfortunately I can't pinpoint at the moment when this happened. I'm 99% certain I was using build from commit 983829cab16833ec932f26724d9420b63536b0a4 when it was working

**Comment by ashinn** [comment]
The earlier fix was insufficient - I started taking a look at the overall distribution, and with even larger numbers, and it was heavily skewed.  This was fixed in `c726273c3b1b731c9b203eb4e2c16df287799da4`.

The above test passes just fine when we use larger (2^64 / 2^65 or 2^164 / 2^165) or smaller (2^62 / 2^63) ranges, but fails at 2^127 / 2^128, so it appears to be a boundary issue.

**Comment by ashinn** [comment]
Actually, although your test still worked for all of those values, the distribution still had gaps in those cases.

Fixed and this time added proper statistical tests to avoid further regressions.
